### DISCUSSION TITLE
SOH

### DISCUSSION THREAD

**Discussion by matthew270492** [nathanmarlor/foxess_modbus#686] [Ideas]
Is it possible to get the SOH entity for this integration? I have a H1 inverter and EP11 battery. It would be good to have the battery state of health as shown on the Fox App.

**Comment by canton7** [comment]
More details please. Which inverter, which entity specifically?

**Reply by matthew270492** [reply]
Updated original post

**Reply by canton7** [reply]
What exactly do you mean by "the SOH entity"? Are you referring to an entity which we show for other inverters?

**Reply by matthew270492** [reply]
I'm not sure if it's available for other inverters. I would like to be able to see the state of health of my batteries in HA on the sensor list. I believe other integrations have it. It may be that it's only available on certain inverters which is why I don't see it.

**Reply by canton7** [reply]
What exactly do you mean by "other integrations"? Other integrations which read foxess modbus registers?

The register list is [here](https://github.com/nathanmarlor/foxess_modbus/files/11628949/Fox.-.AIO.H1.modbus.-.2023.pdf). Is your SOH entity one of those listed?

**Reply by matthew270492** [reply]
![Screenshot_2024-10-02-19-16-31-66_e2d5b3f32b79de1d45acd1fad96fbb0f~2](https://github.com/user-attachments/assets/de114735-54f5-406c-b9a0-711a972570fa)

**Reply by canton7** [reply]
Huh, seems we're actually not mapping that. I'll give it a try next time I'm free

**Reply by canton7** [reply]
See #687

**Reply by matthew270492** [reply]
Brill thank you. Got there in the end 😂

**Reply by MartB** [reply]
@matthew270492 H1 G2 or H1 G1?

**Reply by matthew270492** [reply]
H1 G1 I believe.

**Reply by canton7** [reply]
@matthew270492 I don't seem to be able to poll any of those registers which are highlighted in yellow on my H1-G1. Are you able to read them on yours?

**Comment by matthew270492** [comment]
I'm not sure how to do it if I'm honest. I can see the information on the Fox Cloud both on the app and website. Not sure if that helps sorry.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by canton7** [reply]
Go to Developer Tools, Actions, select FoxESS - Modbus: Read Registers, enter the device and start address, click read.

**Reply by matthew270492** [reply]
I get an error for all highlighted yellow aswell. Even PV Power - 11098 but that is listed as an entity and is working. ( I can't see any other addresses that have just PV power unless it uses a sum of the individual strings)

**Reply by canton7** [reply]
Cool, thanks for checking. It looks like we won't be able to enable this for the H1-G1.

Yeah, we sum the strings ourselves, if I remember correctly.

**Reply by Wiggy93** [reply]
> Cool, thanks for checking. It looks like we won't be able to enable this for the H1-G1.
> 
> Yeah, we sum the strings ourselves, if I remember correctly.

Glad I found this, saved me hours of digging to see if it was in one of the pre-release versions as couldn't find it in foxess_modbus v.1.13.1 suggested in the other thread. Not sure what the difference is between the macq FoxEss Cloud and FoxEss modbus integrations as the cloud version is able to pull the SOH from my H1 G1 inverter
